### PR TITLE
[SofaCommon] Convert tests to Sofa.Testing

### PR DESCRIPTION
**Pull request by fredroy**
#1885

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: SofaKernel/modules/SofaBaseMechanics/SofaBaseMechanics_test/MappingTestCreation.h [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+
+        InVecCoord xp(Np),xp1(Np);
+        InVecDeriv vp(Np),fp(Np),dfp(Np),fp2(Np);
+        OutVecCoord xc(Nc),xc1(Nc);
+        OutVecDeriv vc(Nc),fc(Nc);
+
+        // get position data
+        sofa::testing::copyFromData( xp, inDofs->readPositions() );
+        sofa::testing::copyFromData( xc, outDofs->readPositions() ); // positions and have already been propagated
+        //          cout<<"parent positions xp = "<< xp << endl;
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseMechanics/SofaBaseMechanics_test/MappingTestCreation.h]: clean //cout

File: SofaKernel/modules/SofaImplicitOdeSolver/SofaImplicitOdeSolver_test/CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,11 @@ set(SOURCE_FILES
     
 add_definitions("-DSOFAIMPLICITODESOLVER_TEST_SCENES_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}/scenes\"")
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
-target_link_libraries(${PROJECT_NAME} Sofa.Testing SofaTest SofaDeformable SofaImplicitOdeSolver)
+target_link_libraries(${PROJECT_NAME} Sofa.Testing SofaDeformable SofaImplicitOdeSolver SofaBaseMechanics SofaBaseUtils SofaBoundaryCondition SofaEigen2Solver SceneCreator)
```
Comment (epernod) [inline SofaKernel/modules/SofaImplicitOdeSolver/SofaImplicitOdeSolver_test/CMakeLists.txt:14]: SofaEigen2Solver for which component?

File: SofaKernel/modules/SofaBaseMechanics/SofaBaseMechanics_test/MappingTestCreation.h (hunk 1 above)
Comment (fredroy) [reply SofaKernel/modules/SofaBaseMechanics/SofaBaseMechanics_test/MappingTestCreation.h]: Suggestion please 🤠

File: SofaKernel/modules/SofaImplicitOdeSolver/SofaImplicitOdeSolver_test/CMakeLists.txt (hunk 2 above)
Comment (fredroy) [reply SofaKernel/modules/SofaImplicitOdeSolver/SofaImplicitOdeSolver_test/CMakeLists.txt:14]: EigenBaseSparseMatrix somewhere

**Pull request merged**